yakut/cmd/file_client/_cmd.py

```diff
@@ -0,0 +1,443 @@
+        prog = ProgressReporter()
+        for nid in node_ids_list:
+            fc = FileClient2(node, nid, response_timeout=timeout)
+            prog(f"Touching on node {nid}")
+
+            try:
+                await fc.touch(str(path))
+                _logger.info("Touched %r on node %r", path, nid)
+            except Exception as e:
+                show_error(f"Error touching {path} on node {nid}: {e}")
+                error = True
+
+    return EXIT_CODE_UNSUCCESSFUL if error else 0
+
+
+@file_client.command()
+@click.argument("node_ids", type=parse_int_set)
+@click.argument("path", default="")
+@click.option(
+    "--timeout",
+    "-T",
+    type=float,
+    default=pycyphal.presentation.DEFAULT_SERVICE_REQUEST_TIMEOUT,
+    show_default=True,
+    metavar="SECONDS",
+    help="Service response timeout.",
+)
+@yakut.pass_purser
+@yakut.asynchronous(interrupted_ok=True)
+async def rm(
+    purser: yakut.Purser,
+    node_ids: set[int] | int,
+    path: Path,
+    timeout: float,
+) -> None:
+    """
+    Remove a file or directory on remote node(s) using the standard Cyphal file service.
+    """
+    from pycyphal.application.file import FileClient2
+
+    _logger.debug("node_ids=%r, path=%r, timeout=%r", node_ids, path, timeout)
+    node_ids_list = list(sorted(node_ids)) if isinstance(node_ids, set) else [node_ids]
+    assert isinstance(node_ids_list, list) and all(isinstance(x, int) for x in node_ids_list)
+
+    error = False
+    with purser.get_node("file_client_rm", allow_anonymous=False) as node:
+        prog = ProgressReporter()
+        for nid in node_ids_list:
+            fc = FileClient2(node, nid, response_timeout=timeout)
+            prog(f"Remove from node {nid}")
+
+            try:
+                await fc.remove(str(path))
+                _logger.info("Removed path %r on node %r", path, nid)
+            except FileNotFoundError:
+                show_warning(f"Path {path} not found on node {nid}")
+            except Exception as e:
+                show_error(f"Error removing {path} on node {nid}: {e}")
+                error = True
+
+    return EXIT_CODE_UNSUCCESSFUL if error else 0
+
+
+@file_client.command()
+@click.argument("node_id", type=int)
+@click.argument("src")
+@click.argument("dst", required=False)
+@click.option(
+    "--timeout",
+    "-T",
+    type=float,
+    default=pycyphal.presentation.DEFAULT_SERVICE_REQUEST_TIMEOUT,
+    show_default=True,
+    metavar="SECONDS",
+    help="Service response timeout.",
+)
+@yakut.pass_purser
+@yakut.asynchronous(interrupted_ok=True)
+async def read(
+    purser: yakut.Purser,
+    node_id: int,
+    src: str,
+    dst: str | None,
+    timeout: float,
+) -> None:
+    """
+    Read a file from a remote node using the standard Cyphal file service.
+    """
+    from pycyphal.application.file import FileClient2
+
+    src = PurePosixPath(src)
+    dst = Path(dst) if dst else Path(src.name)
```

Comment:
I think the default behavior should be to stream the data into stdout in binary mode (`sys.stdout.buffer.write(...)`), because this is the unix way and is probably expected by default.

Reply:
I would argue that `cp src dst` is a more unix way than `cat src > dst` for moving files. Also, when using pipes the progress can not be displayed, though correct me on this.